RELEASE/scripts/autoscend/auto_pre_adv.ash

```diff
@@ -158,11 +158,20 @@ void handlePreAdventure(location place)
 
 	if (isActuallyEd())
 	{
-		if((zone_combatMod(place)._int < combat_rate_modifier()) && (have_effect($effect[Shelter Of Shed]) == 0) && auto_have_skill($skill[Shelter Of Shed]))
+		// make sure we have enough MP to cast our most expensive spells
+		// Wrath of Ra (yellow ray) is 40 MP, Curse of Stench (sniff) is 35 MP & Curse of Vacation (banish) is 30 MP.
+		acquireMP(40, 1000);
+		// ensure we can cast at least Fist of the Mummy or Storm of the Scarab.
+		// so we don't waste adventures when we can't actually kill a monster.
+		acquireMP(8, 0);
+
+		if (my_hp() == 0)
 		{
-			acquireMP(25, my_meat());
+			// the game doesn't let you adventure if you have no HP even though Ed
+			// gets a full heal when he goes to the underworld
+			// only necessary if a non-combat puts you on 0 HP.
+			acquireHP();
```

Comment:
I changed acquireHP in auto_restore to only do stuff for Ed if you're on 0 HP and return false otherwise to handle wasting resources for HP as I noticed it was using up linen bandages when it didn't need to due to the acquireHP calls sprinkled throughout the quest functions.

Ed is an "avatar" type class so he doesn't have access to cocoon etc. so wasting MP on healing is never an issue for him. It's actually doubly impossible because he can only get HP from visiting the Underworld which is free the first 3 times in combat or from using linen bandages, stuff like scroll of drastic healing does nothing for him too.

Reply:
Ah see, never played ed, might give it a try here soon, it sounds interesting. You could also update auto_restoration.txt data file to remove the ed healing options if youd like. Sounds like theres not much point in having them in there.

Reply:
Ah see, never played ed, might give it a try here soon, it sounds interesting. Should we just remove ed healing items as options? Youd just need to update data/auto_restoration.txt and remove the ed specific items. Sounds like theres not much point in having them in there.

Reply:
I wouldn't remove them. acquireHP/MP work very well for Ed now, better than they ever have in my experience. If acquireHP tries to use something that won't give Ed HP (as I've seen it do with scrolls of drastic healing) it just keeps trying until it gets to the linen bandages. He has no restrictions on MP use though so it's nice to see Mountain Stream Soda being used when we have it on hand (old implementation never did that which annoyed me constantly) and it buys up the Doc Galaktik stuff when we don't have Ed's own stuff on hand.

The only Ed specific stuff we might want to support would be going to the Underworld (there's a portal in Ed's "campground" which is actually the Pyramid in the desert) when we don't have any linen bandages and we're on 0 HP. If we do that outside of combat it costs an adventure to leave the Underworld hence it's the option of final resort. There are very few places outside of combat we can end up on 0 HP though so I'd ignore this for now until I get support for mining for ores added properly.